### ISSUE TITLE
If libssl1.0.0 is missing, Prisma fails silently (`Error: Migration engine error: `)

### ISSUE THREAD

**Issue by dav1app** [prisma/prisma#13165]
### Bug description

```
Error: Error: Migration engine error:

    at canConnectToDatabase (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:52238:19)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ensureDatabaseExists (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:110225:22)
    at async Object.parse (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:112233:28)
    at async main (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:115680:18)
 ```
 
 This was the message shown using DEBUG=* in Prisma. After I injected a `console.log()`, the error is finally visible.
 
 
![image](https://user-images.githubusercontent.com/3607502/166745550-142ce7ad-735d-4a19-b8d3-40c6b5ee3e46.png)

```
Error: Command failed with exit code 127: /home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/@prisma/engines/migration-engine-debian-openssl-1.0.x cli --datasource <REDACTED> can-connect-to-database
/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/@prisma/engines/migration-engine-debian-openssl-1.0.x: error while loading shared libraries: libssl.so.1.0.0: cannot open shared object file: No such file or directory
    at makeError (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:3507:17)
    at handlePromise (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:4240:33)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async execaCommand (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:52319:16)
    at async canConnectToDatabase (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:52218:9)
    at async ensureDatabaseExists (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:110225:22)
    at async Object.parse (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:112233:28)
    at async main (/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/prisma/build/index.js:115680:18) {
  shortMessage: 'Command failed with exit code 127: /home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/@prisma/engines/migration-engine-debian-openssl-1.0.x cli --datasource postgresql://postgres:root@localhost:5432/postgres?schema=public can-connect-to-database',
  command: '/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/@prisma/engines/migration-engine-debian-openssl-1.0.x cli --datasource postgresql://postgres:root@localhost:5432/postgres?schema=public can-connect-to-database',
  escapedCommand: '"/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/@prisma/engines/migration-engine-debian-openssl-1.0.x" cli --datasource "postgresql://postgres:root@localhost:5432/postgres?schema=public" can-connect-to-database',
  exitCode: 127,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: '/home/davi/Projects/quarta-casa/Repos/fisher-backend/node_modules/@prisma/engines/migration-engine-debian-openssl-1.0.x: error while loading shared libraries: libssl.so.1.0.0: cannot open shared object file: No such file or directory',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
```
### How to reproduce

1. Install libssl1.0.2u on Debian and remove libssl1.0.0_*. 
2. Run `prisma generate`.

# Fix

Debian fans will hate me for this

Go to http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0 and install any redirection package you want (depending on what you use (my case was [libssl1.0.0_1.0.2n-1ubuntu5.9_amd64.deb](http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.9_amd64.deb)) 

### Expected behavior

- A more consistent version of `libquery_engine-debian-openssl-1.0.x.so.node` that preferably identify if `libssl1.0.0` or `libssl` itself exists.

### Prisma information

No need to. 

### Environment & setup

- Debian 11 (bullseye) with backports.
- Node 14,16,17

### Prisma Version

from 
```
        "prisma": "^3.9.2",
        "@prisma/client": "^3.9.2",
```
to
```
"prisma": "3.13.0-dev.29",
"@prisma/client": "3.13.0-dev.29",
```

**Comment by janpio** [comment]
Can you please check your setup again @dav1app (without your OpenSSL workaround/fix) with 4.10.1 and report if the error is more useful now at least? That would be amazing.

I opened a new issue for the general problem of "Migration engine error: " being a useless error without the details of the exception: https://github.com/prisma/prisma/issues/17892

**Comment by aqrln** [comment]
Since there was no follow up and this issue is not actionable without additional details, I'll go ahead and close it.

From the information in the issue description it doesn't look like Prisma is doing anything wrong or unexpected, and this looks like unsupported setup. However, the information is lacking about this setup (e.g. the actual soname of the OpenSSL library, why is there no `libssl.so.1.0.0` symlink etc) and whether there are any systems affected in their default configuration for us to be able to support this.

Please let us know if we should reconsider this by leaving a comment or opening a new issue.
